include/oneapi/dpl/internal/random_impl/linear_congruential_engine.h

```diff
@@ -64,24 +65,29 @@ class linear_congruential_engine
     seed(scalar_type __seed = default_seed)
     {
         // Engine initalization
-        init<internal::type_traits_t<result_type>::num_elems, increment>(__seed);
+        init<internal::type_traits_t<result_type>::num_elems>(__seed);
     }
 
     // Discard procedure
     void
     discard(unsigned long long __num_to_skip)
     {
         // Skipping sequence
-        skip_seq<internal::type_traits_t<result_type>::num_elems, increment>(__num_to_skip);
+        if (__num_to_skip == 0)
+            return;
+        constexpr bool flag = (increment == 0) && (modulus < ::std::numeric_limits<::std::uint32_t>::max()) &&
```

Comment:
Does it make sense to move this flag to a private member of the class, smth like skip_opt_flag, or it would negatively affect perf as we would have an additional field?

Reply:
Discussed offline - agreed not to add such flag as a class member